packages/bridge/src/bridges/matic-bridge.ts

```diff
@@ -0,0 +1,424 @@
+  private parentNet: NetworkConfig | undefined
+
+  constructor(private conf: MaticBridgeConf) {}
+
+  name(): string {
+    return "Matic bridge"
+  }
+
+  id(): string {
+    return `matic-${this.conf.parentId}-${this.conf.maticId}`
+  }
+
+  connect(networks: NetworkConfig[]): MaticBridge {
+    this.maticNet = networks.find((n) => n.chainId === this.conf.maticId)
+    this.parentNet = networks.find((n) => n.chainId === this.conf.parentId)
+    return this
+  }
+
+  isDeposit(from: NetworkConfig, to: NetworkConfig): boolean {
+    return from.chainId === this.conf.parentId && to.chainId === this.conf.maticId
+  }
+
+  isWithdraw(from: NetworkConfig, to: NetworkConfig): boolean {
+    return from.chainId === this.conf.maticId && to.chainId === this.conf.parentId
+  }
+
+  isMaticToken(token: string): boolean {
+    return token.toLowerCase() === this.conf.maticToken
+  }
+
+  get posClient() {
+    return new MaticPOSClient({
+      network: this.conf.networkName,
+      version: this.conf.sdkVersion,
+      parentProvider: this.parentNet?.rpcUrl,
+      maticProvider: this.maticNet?.rpcUrl
+    })
+  }
+
+  async lastStateId(): Promise<ethers.BigNumber> {
+    const contractInstance = new ethers.Contract(
+      "0x0000000000000000000000000000000000001001",
+      [
+        {
+          constant: true,
+          inputs: [],
+          name: "lastStateId",
+          outputs: [
+            {
+              name: "",
+              type: "uint256"
+            }
+          ],
+          payable: false,
+          stateMutability: "view",
+          type: "function"
+        }
+      ],
+      new ethers.providers.JsonRpcProvider(this.maticNet.rpcUrl)
+    )
+
+    return ethers.BigNumber.from((await contractInstance.functions.lastStateId())[0])
+  }
+
+  async getMoves(wallet: string, from: providers.BlockTag = 0, to: providers.BlockTag = "latest"): Promise<Move[]> {
+    return flatten(await Promise.all([
+      this._depositsNative(wallet, from, to),
+      this._depositsERC20(wallet, from, to),
+      this._withdrawsERC20(wallet, from, to),
+      this._depositsERC20Plasma(wallet, from, to)
+    ].map((p) => safeSolve(p, []))))
+  }
+
+  async supportsNative(from: NetworkConfig, to: NetworkConfig): Promise<boolean> {
+    return this.isWithdraw(from, to) // || this.isDeposit(from, to)
+  }
+
+  async estimateNative(from: NetworkConfig, to: NetworkConfig): Promise<MoveEstimate | undefined> {
+    if (this.isDeposit(from, to)) {
+      return {
+        crossTime: MaticBridge.POS_TIME_DEPOSIT,
+        steps: 1
+      }
+    }
+
+    if (this.isWithdraw(from, to)) {
+      return undefined
+
+      // TODO: Re-enable handle complete
+      // return {
+      //   crossTime: MaticBridge.PLASMA_TIME_WITHDRAW,
+      //   steps: 2
+      // }
+    }
+
+    return undefined
+  }
+
+  async moveNative(from: NetworkConfig, to: NetworkConfig, dest: string, amount: BigNumberish): Promise<providers.TransactionRequest[]> {
```

Comment:
`Native` is meant to be the native coin of the `from` chain, when going to Matic to Mainnet this token is Matic, but the other way around it becomes `ETH`. I called it `native` because each chain may use it's own balance to represent any token (for example DAI in xDAI).

Reply:
Ah, I see. Makes sense then.